server/migrations/1620672944865_update-triggers.js

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable camelcase */
+
+const { collections } = require('../db');
+
+exports.up = async (pgm) => {
+  await pgm.dropTrigger(collections.EMPLOYER_SITES, 'update_postal_code_s', { ifExists: true });
+  await pgm.createTrigger(collections.EMPLOYER_SITES, 'update_postal_code_s', {
+    when: 'BEFORE',
+    operation: ['INSERT', 'UPDATE OF body'],
+    language: 'plpgsql',
+    level: 'ROW',
+    replace: true,
+  },
+  `
+  DECLARE
+   loc record;
+   participant record;
+  BEGIN
+    IF NEW.body->>'postalCode' = OLD.body->>'postalCode' THEN
+      RETURN NEW;
+    END IF;
+
+    SELECT longitude::text,latitude::text INTO loc FROM geocodes WHERE REPLACE(postal_code, ' ', '')=REPLACE(NEW.body->>'postalCode', ' ', '');
+
+    IF NOT FOUND THEN
+      SELECT longitude::text,latitude::text INTO loc FROM geocodes WHERE SUBSTRING(REPLACE(postal_code, ' ', ''),1,5)=SUBSTRING(REPLACE(NEW.body->>'postalCode', ' ', ''),1,5);
+    END IF;
+
+    IF NOT FOUND THEN
+      IF NEW.body->>'location' IS NOT NULL THEN
+        NEW.body = NEW.body::jsonb - 'location';
+        DELETE FROM participants_distance where site_id = OLD.id;
```

Comment:
Is this logic wrong? If the new location is not null, we delete and remove location?

Also it seems strange to check for NEW.body and then use OLD.id

Reply:
OLD.id and NEW.id are semantically equal. The logic overall states that if the postal code was updated with a value that wasn't found in the geocodes database, check if the new entry is holding a location value. If it is, then it is no longer relevant and we remove it along with the related entries in the distance table.